### DISCUSSION TITLE
Unused Fields within Visuals

### DISCUSSION THREAD

**Discussion by swolfe2** [sql-bi/Bravo#598] [Ideas]
Hello SQL BI, 

Thank you all for providing such a robust product as Bravo for Power BI. We love the ability to quickly identify unused fields in the data model. However, it appears that it does not take into account when fields are used in a visual and not within a measure/relationship. My team is curious as to why it does not recognize fields used in a visual like some [other tools do](https://en.brunner.bi/measurekiller), and what kind of change management would it take to implement such a feature? 

Thanks!

**Comment by marcosqlbi** [comment]
Supporting such analysis is definitely on our to-do list. As usual, resources (time) are limited. If you are willing to contribute (that's why Bravo is open source!), we can help you locate the area where this would be integrated.
Ideally:
- it should work when connected to PBI Desktop - but the file could not be aligned to the model (we can only see the last saved file) - locating the file requires some hack with Win32 APIs (we don't know the file name most of the times)
- probably it cannot work when connected to PBI Service - but if/when it's possible to download the reports connected, ideally we should scan all of them

**Comment by albertospelta** [comment]
Hi @swolfe2 Yes, the current version only identifies columns used in measures and relationships of the model and to obtain this information we use the DMV `DISCOVER_CALC_DEPENDENCY` that allows to list all the dependencies between objects in the model.

As Marco said, detecting columns referenced in visuals is definitely on our roadmap but we currently don't have a timeline for when we would begin this work. The implementation will require extraction of the contents of a PBIX file and its analysis, for which we could probably rely on libraries such as [pbi-tools](https://github.com/pbi-tools/pbi-tools), which already have support for this.

We will be able to provide more details as we start working on it but, in the meantime, please be sure to follow #271  which we will use to track this feature.
